.github/workflows/ci.yml

```diff
@@ -551,3 +551,12 @@ jobs:
       - name: Test installed mommy
         shell: msys2 {0}
         run: make system=1 test
+
+  test-nixpkgs:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/magic-nix-cache-action@main
+      - run: |
+          nix build --impure --expr '(import <nixpkgs> {}).mommy.overrideAttrs (prev: { src = ./.; ${if prev.version == "1.2.3" then "patches" else null} = []; })'
```

Comment:
> we also added CI because i wanted to add to the "mommy is tested on …" line :P

yay!

if i understand correctly, you apply the necessary patches (i.e. your pr) for release v1.2.3, and for other versions you don't. makes sense. i guess this line can be simplified to something like `nix build --impure --expr '(import <nixpkgs> {}).mommy'` once there has been a release after v1.2.3?

> the ci [seems to work too,](https://github.com/ckiee/mommy/actions/runs/5985457606/job/16237573451) (netbsd fails tho!)

yes, the netbsd issue was fixed in 29683780467972043625e8978ca4c92783505e36. you can rebase if you want, but i'm fine with ignoring it since the fix will be applied when merging anyway.

> but it's a bit scary because the output doesn't seem to be displayed, but running it locally shows it is running mommy's tests inside nix, so it should be fine.

that is scary. did you verify the build fails if the tests fail? if so, i'm fine with it.

Reply:
> > we also added CI because i wanted to add to the "mommy is tested on …" line :P
> 
> yay!
> 
> if i understand correctly, you apply the necessary patches (i.e. your pr) for release v1.2.3, and for other versions you don't. makes sense. i guess this line can be simplified to something like `nix build --impure --expr '(import <nixpkgs> {}).mommy'` once there has been a release after v1.2.3?
> 

yes, very close!! it'd be:
```nix
# nix build --impure --expr \
  '(import <nixpkgs> {}).mommy.overrideAttrs (prev: { src = ./.; })
```

which overrides `mommy`'s source directory to be the current checkout we're testing, instead of nixpkgs' mommy (:

> > the ci [seems to work too,](https://github.com/ckiee/mommy/actions/runs/5985457606/job/16237573451) (netbsd fails tho!)
> 
> yes, the netbsd issue was fixed in [2968378](https://github.com/FWDekker/mommy/commit/29683780467972043625e8978ca4c92783505e36). you can rebase if you want, but i'm fine with ignoring it since the fix will be applied when merging anyway.
> 

mraow! done

> > but it's a bit scary because the output doesn't seem to be displayed, but running it locally shows it is running mommy's tests inside nix, so it should be fine.
> 
> that is scary. did you verify the build fails if the tests fail? if so, i'm fine with it.

i didn't try it- we have a lot of blind confidence in nix, but here, it actually emits a very nice error (: 

```diff
modified   src/main/sh/mommy
@@ -239,7 +239,7 @@ if [ -n "$MOMMY_OPT_HELP" ]; then
     exit "$status"
 elif [ -n "$MOMMY_OPT_VERSION" ]; then
     echo "mommy, v%%VERSION_NUMBER%%, %%VERSION_DATE%%"
-    exit 0
+    exit 1
 else
     # Run command
     if [ -n "$MOMMY_OPT_EVAL" ]; then
```
![the mommy `nix build` fails with a "shellspec src/test/sh/unit_spec.sh:32 # 1) mommy command-line options -v/--version: version information outputs version information using -v FAILED
" and other similar errors, it's nicely colored](https://github.com/FWDekker/mommy/assets/25263210/c78dce36-d111-4fa4-ad96-951afc1d5840)

so yes (: glad i tried this. here's [the log of a happy run](https://i.ckie.dev/6p6KHqX.txt), if you're curious :3